src/Styleguide/Components/ReadMore2.tsx

```diff
@@ -0,0 +1,100 @@
+import { track } from "Analytics"
+import * as Schema from "Analytics/Schema"
+import React, { Component } from "react"
+import styled from "styled-components"
+import { DisplayProps } from "styled-system"
+
+export interface ReadMoreProps extends DisplayProps {
+  isExpanded?: boolean
+  maxChars?: number
+  onReadMoreClicked?: () => void
+  text: string
+}
+
+export interface ReadMoreState {
+  isExpanded: boolean
+}
+
+export class ReadMore extends Component<ReadMoreProps, ReadMoreState> {
+  state = {
+    isExpanded: true,
+  }
+
+  static defaultProps = {
+    isExpanded: false,
+    maxChars: Infinity,
+  }
+
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      isExpanded: props.isExpanded,
+    }
+  }
+
+  @track({
+    action_type: Schema.ActionType.Click,
+    subject: "Read more",
+  })
+  expandText() {
+    this.setState(
+      {
+        isExpanded: true,
+      },
+      () => {
+        this.props.onReadMoreClicked && this.props.onReadMoreClicked()
+      }
+    )
+  }
+
+  render() {
+    const { isExpanded } = this.state
+    const { text } = this.props
+    const textSlice = isExpanded ? text : text.substr(0, this.props.maxChars)
+
+    return (
+      <Container onClick={this.expandText.bind(this)} isExpanded>
```

Comment:
rather than the bind, is it more idiomatic now to do `onClick={() => this.expandText()}` ?

Reply:
This is required because the method is bound to a `@track` function